### PR TITLE
Support setMailboxes (issue #20)

### PR DESCRIPTION
**Pull request by dbenchi**

### CODE REVIEW

File: lib/API.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ export default {
   Thread: require('./models/Thread'),
   MessagesSet: require('./models/MessagesSet'),
   MailboxRole: require('./models/MailboxRole'),
+  MailboxesSet: require('./models/MailboxesSet'),
```
Comment (ddolcimascolo) [inline lib/API.js:65]: You need to add an `@property` jsdoc tag above.

File: lib/client/Client.js [hunk 2]
```diff
@@ -221,6 +222,96 @@ export default class Client {
...
+   */
+  setMailboxes(options) {
+    return this._jmapRequest('setMailboxes', options);
+  }
+
+  /**
+   * Creates a mailbox  by sending a _setMailboxes_ JMAP request.<br />
+   *
+   * @param name {String} The name of the mailbox to create.
+   * @param parentId {String} The id of the parent of the mailbox to create.
```
Comment (ddolcimascolo) [inline lib/client/Client.js]: It's an optional parameter, describe it as such: `[parentId=null]`

File: lib/client/Client.js [hunk 3]
```diff
@@ -221,6 +222,96 @@ export default class Client {
...
+   *
+   * @return {Promise} A {@link Promise} that eventually resolves to a {@link MailboxesSet}.
```
Comment (ddolcimascolo) [inline lib/client/Client.js]: Wrong, you're resolving with a `CreatedMailboxAck`, a new Class that you need to add. See `CreatedMessageAck` for a similar use case.

File: lib/models/Mailbox.js [hunk 4]
```diff
@@ -83,6 +83,30 @@ export default class Mailbox extends Model {
   }
 
   /**
+   * Update this {@link Mailbox}.
+   *
+   * @param options {Object} The attribute to be updated in this {@link Mailbox}.
```
Comment (ddolcimascolo) [inline lib/models/Mailbox.js:88]: Please document that we can only pass `options.name` and `options.parentId`. See jsDoc documentation for details, or other examples in the code.

File: lib/API.js (hunk 1 above)
Comment (dbenchi) [inline lib/API.js:65]: Done

File: lib/models/Mailbox.js (hunk 4 above)
Comment (dbenchi) [inline lib/models/Mailbox.js:88]: Done

File: lib/client/Client.js (hunk 2 above)
Comment (dbenchi) [inline lib/client/Client.js]: Done

File: lib/client/Client.js (hunk 3 above)
Comment (dbenchi) [inline lib/client/Client.js]: Done

File: lib/client/Client.js [hunk 5]
```diff
@@ -221,6 +223,96 @@ export default class Client {
...
+    var clientId = this._generateClientId();
+
+    return this.setMailboxes({
+      create: {
+        [clientId]: {
+          name: name,
+          parentId: parentId
+        }
+      }
+    }).then(function(mailboxesSet) {
```
Comment (thomas-hilaire) [inline lib/client/Client.js]: use arrow functions (everywhere)

Comment (dbenchi) [inline lib/client/Client.js]: Done

**Comment by ddolcimascolo** [comment]
:+1:

**Comment by thomas-hilaire** [comment]
:+1:

**Pull request merged**